Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,359 @@
+    previous_build_in_queue = ''
+    if my_place_in_the_queue > 0:
+        previous_build_in_queue = sorted_builds_in_queue[my_place_in_the_queue - 1].get('name')
+    return my_place_in_the_queue, previous_build_in_queue
+
+
+def try_to_lock_machine(storage_bucket: any, machine: str, machines_locks: list, gitlab_status_token: str,
+                        gcs_locks_path: str, job_id: str) -> str:
+    """
+    try to lock machine for the job
+    Args:
+        storage_bucket(google.cloud.storage.bucket.Bucket): google storage bucket where lock machine is stored.
+        machine(str): the machine to lock.
+        machines_locks(srt): all the exiting lock files.
+        gitlab_status_token(str): the gitlab token.
+        gcs_locks_path(str): the lock repository name.
+        job_id(str): the job id to check.
+
+    Returns: the machine name if locked.
+
+    """
+    lock_machine_name = ''
+    job_id_of_the_existing_lock = next((d['job_id'] for d in machines_locks if d["machine_name"] == machine), None)
+    if job_id_of_the_existing_lock:
+        logging.debug(f'There is a lock file for job id: {job_id_of_the_existing_lock}')
+        job_id_of_the_existing_lock_status = check_job_status(gitlab_status_token, job_id_of_the_existing_lock)
+        logging.debug(f'the status of job id: {job_id_of_the_existing_lock} is: {job_id_of_the_existing_lock_status}')
+        if job_id_of_the_existing_lock_status != 'running':
+            # machine found! removing the not relevant lock and create a now one
+            remove_machine_lock_file(storage_bucket, gcs_locks_path, machine, job_id_of_the_existing_lock)
+        else:
+            return lock_machine_name
+    else:
+        # machine found! create lock file
+        logging.info('There is no existing lock file')
+    logging.info(f'Locking machine {machine}')
+    lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
+    lock_machine_name = machine
+    return lock_machine_name
+
+
+def get_and_lock_all_needed_machines(storage_client, storage_bucket, list_machines, gcs_locks_path,
+                                     number_machines_to_lock, job_id, gitlab_status_token):
+    """
+    get the requested machines and locked them to the job-id.
+    Args:
+        storage_client(storage.Client): The GCP storage client.
+        storage_bucket(google.cloud.storage.bucket.Bucket): google storage bucket where lock machine is stored.
+        list_machines (list): all the exiting machines.
+        gcs_locks_path(str): the lock repository name.
+        number_machines_to_lock(int): the number of the requested machines.
+        job_id(str): the job id to lock.
+        gitlab_status_token(str): the gitlab token.
+
+    Returns: the machine name if locked.
+    """
+
+    logging.debug('getting all machines lock files')
+    machines_locks = get_machines_locks_details(storage_client, 'xsoar-ci-artifacts',
```

Comment:
Please use as constant:
LOCKS_BUCKET = 'xsoar-ci-artifacts'

```suggestion
    machines_locks = get_machines_locks_details(storage_client, LOCKS_BUCKET,
```